### PR TITLE
util-macros: ensure url_for_version works for older versions

### PR DESCRIPTION
**Pull request by wdconinc**
This PR should fix the issue raised in https://github.com/spack/spack/pull/44388#discussion_r1617785674.

### CODE REVIEW

**Comment by wdconinc** [comment]
After 2d5bf3534e455d191595fd082d613e926fff7528 this pulls the right URL now. Likely will have a style flag.
```
root@codespaces-60ee7e:/workspaces/spack# spack install --fresh --no-cache util-macros@1.19.3
...
==> Installing util-macros-1.19.3-y2yjfowwf3dpdbw53pdu6ros4c5l7csl [4/4]
==> Fetching https://www.x.org/archive/individual/util/util-macros-1.19.3.tar.bz2
==> No patches needed for util-macros
==> util-macros: Executing phase: 'autoreconf'
==> util-macros: Executing phase: 'configure'
==> util-macros: Executing phase: 'build'
==> util-macros: Executing phase: 'install'
==> util-macros: Successfully installed util-macros-1.19.3-y2yjfowwf3dpdbw53pdu6ros4c5l7csl
  Stage: 0.97s.  Autoreconf: 0.00s.  Configure: 1.29s.  Build: 0.01s.  Install: 0.04s.  Post-install: 0.03s.  Total: 2.42s
[+] /workspaces/spack/opt/spack/linux-ubuntu20.04-x86_64_v3/gcc-11.4.0/util-macros-1.19.3-y2yjfowwf3dpdbw53pdu6ros4c5l7csl
```

@tldahlgren I think this is testable now. Above was in a clean codespaces, but I had to remove the builtion spack-public mirror (`rm etc/spack/default/mirorrs.yaml`); that's what had interfered before.

File: var/spack/repos/builtin/packages/util-macros/package.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class UtilMacros(AutotoolsPackage, XorgPackage):
     # note: url_for_version can only return a single url, no mirrors
     def url_for_version(self, version):
         if self.spec.satisfies("@:1.19"):
-            return self.urls[0].replace("xz", "bz2")
+            return spack.url.substitute_version(self.urls[0].replace("xz", "bz2"), version)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/util-macros/package.py]: This method should be moved to below the directives, say around line 32.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/util-macros/package.py]: 🤔 This does not appear to work with `spack checksum`.  I can only confirm `1.20.1` and `spack checksum util-macros` only produces checksums for that and `1.20.0`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Review by tldahlgren — APPROVED** [review]
I *can* stage the versions not covered by 'spack checksum', which I find *very* annoying but haven't yet resolved.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Correction: I would still like you to move the method so it is the first method in the package.

**Comment by wdconinc** [comment]
Done.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**